### PR TITLE
iterators_task_2 Chyruk

### PR DESCRIPTION
**Pull request by ChNokia**

### CODE REVIEW

File: iterators_task_2/student/tests/unit_tests/src/maintest.cpp [hunk 1]
```diff
@@ -144,14 +144,14 @@ TEST(Properties, Traits)
...
-    static_assert (std::is_assignable<accumulator,
-                                      rvalue_container>::value,
-                   "Shall support container assign (rvalue)");
-
-    using another_container = std::deque<int>;
-    using const_ref_another = std::add_lvalue_reference<std::add_const<another_container>::type>::type;
-    static_assert (std::is_assignable<accumulator,
-                                      const_ref_another>::value,
-                   "Shall support another container assign (lvalue)");
+//    using rvalue_container = std::add_rvalue_reference<container>::type;
```
Comment (smay1613) [inline iterators_task_2/student/tests/unit_tests/src/maintest.cpp:147]: why commented?

File: iterators_task_2/student/include/impl.h [hunk 2]
```diff
@@ -2,14 +2,46 @@
...
+
     /** @todo Constructor with container*/
+    accumulator(Container& container) : m_cotainer(container)
+    {}
+
+    accumulator(const accumulator<Container>& other) = default;
+
     /** @todo lvalue writing operator*/
+    template <typename OtherContainer>
+    accumulator& operator=(const OtherContainer& __value) {
```
Comment (smay1613) [inline iterators_task_2/student/include/impl.h:16]: use more descriptive names

File: iterators_task_2/student/include/impl.h [hunk 3]
```diff
@@ -2,14 +2,46 @@
...
-struct accumulator
+struct accumulator : public std::iterator<std::output_iterator_tag, void,void,void,void>
 {
-    /** @todo Iterator traits*/
+
     /** @todo Constructor with container*/
+    accumulator(Container& container) : m_cotainer(container)
+    {}
+
+    accumulator(const accumulator<Container>& other) = default;
```
Comment (smay1613) [inline iterators_task_2/student/include/impl.h:12]: what about assignment operator?

File: iterators_task_2/student/tests/unit_tests/src/maintest.cpp (hunk 1 above)
Comment (ChNokia) [reply iterators_task_2/student/tests/unit_tests/src/maintest.cpp:147]: it's my fault

File: iterators_task_2/student/include/impl.h (hunk 2 above)
Comment (ChNokia) [reply iterators_task_2/student/include/impl.h:16]: what do you mean?(should I use "Container" or "Other"?)

Comment (smay1613) [reply iterators_task_2/student/include/impl.h:16]: I mean "value"
actually, value is not a value, it is a container

**Pull request closed without merging**